### ISSUE TITLE
4.9.0 Release checklist

### ISSUE THREAD

**Issue by compulim** [microsoft/BotFramework-WebChat#2994]
# Checklist

## Build

1. [x] ~Bump [DirectLineJS](https://github.com/microsoft/BotFramework-DirectLineJS.git) to latest version~
   - Currently one of the dependencies which latest version of DLJS use, does not work in browser
   - https://github.com/microsoft/botbuilder-js/pull/2223
1. [x] Bump to Bot Framework SDK release 4.9.0
   - ~https://botbuilder.myget.org/feed/botbuilder-v4-js-daily/package/npm/botbuilder~
   - https://www.npmjs.com/package/botbuilder/v/4.9.0
1. [x] Update `CHANGELOG.md` to mark specific changes in `4.9.0`
1. [x] Run `npm version --no-git-tag-version 4.9.0`
   - [x] Merged into `master`, the PR number is #3151
   - Commit is `6b6caffbcff301a8c75873a47e3a8067bd99c9fa`
   - Do not merge any other unrelated changes after this PR. Any other PR merged, will need to be re-tested
1. [x] Run daily pipeline manually, set "generate release version number" to `true`
   - (This will not push to NPM or CDN)
   - The build number is `130299` and commit is `6b6caffbcff301a8c75873a47e3a8067bd99c9fa`

## Test

> The test should run against the build artifacts from Azure Pipelines.

1. [x] Manual testing on major browsers using `webchat-release-testing`
   - [x] Chrome 81.0.4044.138
   - [x] Edge (Anaheim) 84.0.512.0
   - [x] Edge (UWP) 44.19041.1.0
   - [x] Firefox 76.0.1
   - [x] IE11
   - [x] Macbook Safari 13.0.3 (15608.3.10.1.4)
   - [x] iOS Safari 13.4.1
   - [x] Android Chrome 81.0.4044.138
1. [x] Test specific fixes related to `4.9.0` and previous releases
   
   - [x] #3146
      - Currently not fully testable blocked by AC package

## Release

1. [x] Make sure you are on `master` ~or `qfe`~ branch, run `git status` to check
1. [x] `git pull`
1. [x] Verify `/package.json`, `/package-lock.json`, and `CHANGELOG.md` has a version of `4.9.0`
1. [x] `git log`
   - Verify the latest commit is `6b6caff`
1. [x] `git tag v4.9.0`
1. [x] `git push -u upstream v4.9.0`
   - You do not need to kick off a build again, use the previous build
1. [x] Create a new GitHub release, copy entries from `CHANGELOG.md`
   - [x] Subresource Integrity can be generated by `curl https://cdn.botframework.com/botframework-webchat/4.9.0/webchat.js | openssl dgst -sha384 -binary | openssl base64 -A`
   - [x] Attach assets including 3 JS files and 4 tarballs
      - You can copy the artifacts from `webchat-release-testing/drops`
      - Tarballs download from npmjs
         ```
         curl -L -o botframework-directlinespeech-sdk-4.9.0.tgz https://registry.npmjs.org/botframework-directlinespeech-sdk/-/botframework-directlinespeech-sdk-4.9.0.tgz
         curl -L -o botframework-webchat-4.9.0.tgz https://registry.npmjs.org/botframework-webchat/-/botframework-webchat-4.9.0.tgz
         curl -L -o botframework-webchat-core-4.9.0.tgz https://registry.npmjs.org/botframework-webchat-core/-/botframework-webchat-core-4.9.0.tgz
         curl -L -o botframework-webchat-component-4.9.0.tgz https://registry.npmjs.org/botframework-webchat-component/-/botframework-webchat-component-4.9.0.tgz
         ```
1. [x] Kick off release to NPM
   - The build number is `130299`, release number is `2112` and commit is `6b6caff`
1. [x] Kick off release to CDN
   - The build number is `130299`, release number is `2126` and commit is `6b6caff`
   - Script build number is `131156`
   1. [x] Send reminder email to approvers

## Post-release check

- [x] Clone `webchat-release-testing`
   - `01.create-react-app`
      1. [x] Nuke `01.create-react-app/node_modules`
      1. [x] `npm install botframework-webchat@4.9.0` (just install the bundle package)
   - Others
      -  Using script tags from https://github.com/microsoft/BotFramework-WebChat/releases/tag/v4.9.0, with subresource integrity
         ```
         <script
            crossorigin="anonymous"
            integrity="sha384-"
            src="https://cdn.botframework.com/botframework-webchat/latest/webchat-es5.js"
         ></script>
         ```

## Notification to interesting parties

- [x] Update partner page on Adaptive Cards doc
   - https://docs.microsoft.com/en-us/adaptive-cards/resources/partners
   - PR at https://github.com/MicrosoftDocs/AdaptiveCards/pull/284
- [x] Notify related parties for the following fixes
   
# Changelog

### Added

-  Resolves [#2897](https://github.com/microsoft/BotFramework-WebChat/issues/2897). Moved from JUnit to VSTest reporter with file attachments, by [@compulim](https://github.com/compulim) in PR [#2990](https://github.com/microsoft/BotFramework-WebChat/pull/2990)
-  Added `aria-label` attribute support for default Markdown engine, by [@patniko](https://github.com/patniko) in PR [#3022](https://github.com/microsoft/BotFramework-WebChat/pull/3022)
-  Resolves [#2969](https://github.com/microsoft/BotFramework-WebChat/issues/2969). Support sovereign cloud for Cognitive Services Speech Services, by [@compulim](https://github.com/compulim) in PR [#3040](https://github.com/microsoft/BotFramework-WebChat/pull/3040)
-  Resolves [#2481](https://github.com/microsoft/BotFramework-WebChat/issues/2481). Support selecting different audio input devices for Cognitive Services Speech Services, by [@compulim](https://github.com/compulim) in PR [#3079](https://github.com/microsoft/BotFramework-WebChat/pull/3079)
-  Resolves [#2850](https://github.com/microsoft/BotFramework-WebChat/issues/2850). Added new `useFocus` hook and deprecating `useFocusSendBox` hook, by [@compulim](https://github.com/compulim) in PR [#3123](https://github.com/microsoft/BotFramework-WebChat/pull/3123)
   -  Modify `setFocus` argument of `useTextBoxSubmit` to support `main` and `sendBoxWithoutKeyboard`

### Fixed

-  Fixes [#2989](https://github.com/microsoft/BotFramework-WebChat/issues/2989). Fix `observeOnce` to use ES Observable call pattern, by [@compulim](https://github.com/compulim) in PR [#2993](https://github.com/microsoft/BotFramework-WebChat/pull/2993)
-  Fixes [#3024](https://github.com/microsoft/BotFramework-WebChat/issues/3024). Using bridge package [`markdown-it-attrs-es5`](https://npmjs.com/package/markdown-it-attrs-es5) for consuming [`markdown-it-attrs`](https://npmjs.com/package/markdown-it-attrs) for IE11, by [@compulim](https://github.com/compulim) in PR [#3025](https://github.com/microsoft/BotFramework-WebChat/pull/3025)
-  Fixes [#2818](https://github.com/microsoft/BotFramework-WebChat/issues/2818). Fix user ID is not set when passing to embed as query parameter, by [@p-nagpal](https://github.com/p-nagpal) in PR [#3031](https://github.com/microsoft/BotFramework-WebChat/pull/3031)
-  Fixes [#3026](https://github.com/microsoft/BotFramework-WebChat/issues/3026). Fix link `rel` attribute in the `renderMarkdown` function, by [@tdurnford](https://github.com/tdurnford) in PR [#3033](https://github.com/microsoft/BotFramework-WebChat/pull/3033)
-  Fixes [#2933](https://github.com/microsoft/BotFramework-WebChat/issues/2933). Fix `text` should not be ignored in `messageBack` action in hero card, by [@geea-develop](https://github.com/geea-develop) and [@compulim](https://github.com/compulim) in PR [#3003](https://github.com/microsoft/BotFramework-WebChat/pull/3003)
-  Fixes [#2562](https://github.com/microsoft/BotFramework-WebChat/issues/2562). Fix timestamps should not stop updating, by [@compulim](https://github.com/compulim) in PR [#3066](https://github.com/microsoft/BotFramework-WebChat/pull/3066)
-  Fixes [#2953](https://github.com/microsoft/BotFramework-WebChat/issues/2953). Direct Line Speech should not synthesize when the `speak` property is falsy, by [@compulim](https://github.com/compulim) in PR [#3059](https://github.com/microsoft/BotFramework-WebChat/pull/3059)
-  Fixes [#2876](https://github.com/microsoft/BotFramework-WebChat/issues/2876). `messageBack` and `postBack` should send even if both `text` and `value` is falsy or `undefined`, by [@compulim](https://github.com/compulim) in PR [#3120](https://github.com/microsoft/BotFramework-WebChat/issues/3120)
-  Fixes [#2668](https://github.com/microsoft/BotFramework-WebChat/issues/2668). Disable Web Audio on insecure connections, by [@compulim](https://github.com/compulim) in PR [#3079](https://github.com/microsoft/BotFramework-WebChat/pull/3079)
-  Fixes [#2850](https://github.com/microsoft/BotFramework-WebChat/issues/2850). After click suggested action, should focus to send box without keyboard, by [@compulim](https://github.com/compulim) in PR [#3123](https://github.com/microsoft/BotFramework-WebChat/pull/3123)
-  Fixes [#3133](https://github.com/microsoft/BotFramework-WebChat/issues/3133). Associate ARIA labels with buttons in hero card and Adaptive Cards, by [@compulim](https://github.com/compulim) in PR [#3146](https://github.com/microsoft/BotFramework-WebChat/pull/3146).
   -  Remove browser-based detection from `<ScreenReaderText>` because it is no longer needed.
   -  After stripping Markdown syntax for accessibility labels, cache the result to improve rendering performance.
   -  Skip stripping Markdown for non-Markdown text content.
-  Fixes [#3155](https://github.com/microsoft/BotFramework-WebChat/issues/3155). Patch incoming activities with null fields, by [@compulim](https://github.com/compulim) in PR [#3154](https://github.com/microsoft/BotFramework-WebChat/pull/3154)

### Changed

-  Bumped all dependencies to latest versions, by [@compulim](https://github.com/compulim) in PR [#2985](https://github.com/microsoft/BotFramework-WebChat/pull/2985) and [#3012](https://github.com/microsoft/BotFramework-WebChat/pull/3012)
   -  Development dependencies
      -  Root package
         -  [`@azure/storage-blob@12.1.0`](https://npmjs.com/package/@azure/storage-blob)
         -  [`@babel/plugin-proposal-class-properties@7.8.3`](https://npmjs.com/package/@babel/plugin-proposal-class-properties)
         -  [`@babel/plugin-proposal-object-rest-spread@7.8.3`](https://npmjs.com/package/@babel/plugin-proposal-object-rest-spread)
         -  [`@babel/plugin-transform-runtime@7.8.3`](https://npmjs.com/package/@babel/plugin-transform-runtime)
         -  [`@babel/preset-env@7.8.7`](https://npmjs.com/package/@babel/preset-env)
         -  [`@babel/preset-react@7.8.3`](https://npmjs.com/package/@babel/preset-react)
         -  [`@babel/preset-typescript@7.8.3`](https://npmjs.com/package/@babel/preset-typescript)
         -  [`@babel/runtime@7.8.7`](https://npmjs.com/package/@babel/runtime)
         -  [`babel-jest@25.1.0`](https://npmjs.com/package/babel-jest)
         -  [`concurrently@5.1.0`](https://npmjs.com/package/concurrently)
         -  [`core-js@3.6.4`](https://npmjs.com/package/core-js)
         -  [`cross-env@7.0.2`](https://npmjs.com/package/cross-env)
         -  [`get-port@5.1.1`](https://npmjs.com/package/get-port)
         -  [`husky@4.2.3`](https://npmjs.com/package/husky)
         -  [`jest@25.1.0`](https://npmjs.com/package/jest)
         -  [`jest-image-snapshot@2.12.0`](https://npmjs.com/package/jest-image-snapshot)
         -  [`lerna@3.20.2`](https://npmjs.com/package/lerna)
         -  [`lint-staged@10.1.1`](https://npmjs.com/package/lint-staged)
         -  [`selenium-webdriver@4.0.0-alpha.7`](https://npmjs.com/package/selenium-webdriver)
      -  Other packages
         -  [`@babel/core@7.8.7`](https://npmjs.com/package/@babel/core)
         -  [`@babel/preset-env@7.8.7`](https://npmjs.com/package/@babel/preset-env)
         -  [`babel-jest@25.1.0`](https://npmjs.com/package/babel-jest)
         -  [`babel-plugin-istanbul@6.0.0`](https://npmjs.com/package/babel-plugin-istanbul)
         -  [`concurrently@5.1.0`](https://npmjs.com/package/concurrently)
         -  [`eslint-plugin-prettier@3.1.2`](https://npmjs.com/package/eslint-plugin-prettier)
         -  [`eslint-plugin-react-hooks@2.5.0`](https://npmjs.com/package/eslint-plugin-react-hooks)
         -  [`eslint-plugin-react@7.18.3`](https://npmjs.com/package/eslint-plugin-react)
         -  [`eslint@6.8.0`](https://npmjs.com/package/eslint)
         -  [`terser-webpack-plugin@2.3.5`](https://npmjs.com/package/terser-webpack-plugin)
         -  [`typescript@3.8.3`](https://npmjs.com/package/typescript)
         -  [`webpack-cli@3.3.11`](https://npmjs.com/package/webpack-cli)
         -  [`webpack-stats-plugin@0.3.1`](https://npmjs.com/package/webpack-stats-plugin)
         -  [`webpack@4.42.0`](https://npmjs.com/package/webpack)
   -  Production dependencies
      -  `core`
         -  [`@babel/runtime@7.8.7`](https://npmjs.com/package/@babel/runtime)
         -  [`redux-saga@1.1.3`](https://npmjs.com/package/redux-saga)
      -  `bundle`
         -  [`@babel/runtime@7.8.7`](https://npmjs.com/package/@babel/runtime)
         -  [`core-js@3.6.4`](https://npmjs.com/package/core-js)
         -  [`url-search-params-polyfill@8.0.0`](https://npmjs.com/package/url-search-params-polyfill)
      -  `component`
         -  [`react-redux@7.2.0`](https://npmjs.com/package/react-redux)
         -  [`redux@4.0.5`](https://npmjs.com/package/redux)
      -  `directlinespeech`
         -  [`@babel/runtime@7.8.7`](https://npmjs.com/package/@babel/runtime)
         -  [`core-js@3.6.4`](https://npmjs.com/package/core-js)
      -  `embed`
         -  [`@babel/runtime@7.8.7`](https://npmjs.com/package/@babel/runtime)
         -  [`core-js@3.6.4`](https://npmjs.com/package/core-js)
-  Bumped Chrome Docker image to `3.141.59-zirconium` (Chrome 80.0.3987.106), by [@compulim](https://github.com/compulim) in PR [#2992](https://github.com/microsoft/BotFramework-WebChat/pull/2992)
-  Added `4.8.0` to `embed/servicingPlan.json`, by [@compulim](https://github.com/compulim) in PR [#2986](https://github.com/microsoft/BotFramework-WebChat/pull/2986)
-  Bumped `microsoft-cognitiveservices-speech-sdk@1.10.1` and `web-speech-cognitive-services@6.1.0`, by [@compulim](https://github.com/compulim) in PR [#3040](https://github.com/microsoft/BotFramework-WebChat/pull/3040)
-  Resolved [#2886](https://github.com/microsoft/BotFramework-WebChat/issues/2886) and [#2987](https://github.com/microsoft/BotFramework-WebChat/issue/2987), converged all references of [`microsoft-cognitiveservices-speech-sdk`](https://npmjs.com/package/microsoft-cognitiveservices-speech-sdk) to reduce footprint, by [@compulim](https://github.com/compulim) in PR [#3079](https://github.com/microsoft/BotFramework-WebChat/pull/3079)

## Samples

-  Resolves [#2806](https://github.com/microsoft/BotFramework-WebChat/issues/2806), added [Single sign-on with On Behalf Of Token Authentication](https://webchat-sample-obo.azurewebsites.net/) sample, by [@tdurnford](https://github.com/tdurnford) in [#2865](https://github.com/microsoft/BotFramework-WebChat/pull/2865)
-  Resolves [#2481](https://github.com/microsoft/BotFramework-WebChat/issues/2481), added selectable audio input device sample, by [@compulim](https://github.com/compulim) in PR [#3079](https://github.com/microsoft/BotFramework-WebChat/pull/3079)
